src/internal/connector/exchange/service_iterators_test.go

```diff
@@ -1403,54 +1405,71 @@ func (suite *ServiceIteratorsSuite) TestFilterContainersAndFillCollections_incre
 				"4": {data.MovedState, true},
 				"5": {data.DeletedState, false},
 			},
+			skipForNonDelta: true,
 		},
 	}
 	for _, test := range table {
-		suite.Run(test.name, func() {
-			t := suite.T()
+		for _, canMakeDeltaQueries := range []bool{true, false} {
+			name := test.name
 
-			ctx, flush := tester.NewContext()
-			defer flush()
+			if canMakeDeltaQueries {
+				name += "-delta"
+			} else {
+				if test.skipForNonDelta {
+					continue
+				}
+				name += "-non-delta"
```

Comment:
Instead of silently skipping, lets make this loud:
```suggestion
				name += "-non-delta"
				
				// move this just inside suite.Run, line 1426 or so.
				if test.skipForNonDelta {
					t.Skip("intentionally skipped non-delta case")
				}
```